### PR TITLE
MP Refactoring/Cleanup

### PR DESCRIPTION
**Pull request by bryates**
This PR adds some additional updates to the MP disks (merge after #242). It improves the timing (currently down to 3.58 ns) and removes some unused variables.

### CODE REVIEW

File: TrackletAlgorithm/MatchEngineUnit.h [hunk 1]
```diff
@@ -448,6 +436,10 @@ inline void advance() {
  ProjectionRouterBuffer<VMProjType, AllProjectionType> projbuffer_;
  ap_uint<kNBits_MemAddr> projseq_;
  bool isPSseed_;
+ ap_uint<1> usefirstMinus;
```
Comment (aryd) [inline TrackletAlgorithm/MatchEngineUnit.h:439]: Use trailing _ for member data.

Comment (aryd) [reply TrackletAlgorithm/MatchEngineUnit.h:439]: Actually - as far as I can tell these are only used in one method, init(), so they should not be class variables, but local variables in the init method.

Comment (bryates) [reply TrackletAlgorithm/MatchEngineUnit.h]: Good catch, I've moved them.

**Review by aryd — APPROVED** [review]
I'm happy and set this as approved so that we can merge this.

**Comment by aryd** [comment]
We have an issue with MP_L2PHIC not giving the right result, but it was caused in an earlier PR with new test vectors so will leave this as is and merge this PR.

**Pull request merged**